### PR TITLE
feat(annotations): add global annotations

### PR DESCRIPTION
**Pull request by GMartinez-Sisti**
## what

- Allow adding global annotations to created resources
- Linted some files to make reading easier

## why

Requested feature.

## tests

Templates were generated against the main branch and this branch and compared to ensure there were no extra changes

```sh
→ diff --color ci-init-config-values-main.yaml ci-init-config-values-new.yaml
```
```diff
113a114
>
127,129c128,130
<         checksum/config: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
<         checksum/repo-config: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
<         checksum/init-config: da4123f44ff227abfe55354cd7872ec561edf6bc58885cca89bea77d6f533a21
---
>         checksum/config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
>         checksum/repo-config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
>         checksum/init-config: e035a1806c7275d6e396d5c7d9b2b0bd51252a161e68ef3e652ee0fc5e6576d9
```

```sh
→ diff --color ci-values-main.yaml ci-values-new.yaml
```
``` diff
77a78
>
91,92c92,93
<         checksum/config: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
<         checksum/repo-config: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
---
>         checksum/config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
>         checksum/repo-config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
```

Then I tested specifying an annotation and a secret:

```sh
→ helm template atlantis ./charts/atlantis/ --set api.secret=aaa --set extraAnnotations.aaa=bbb | diff --color old.yaml 
```
```diff
-12a13,30
>   annotations:
>     aaa: bbb
> ---
> # Source: atlantis/templates/secret-api.yaml
> apiVersion: v1
> kind: Secret
> metadata:
>   name: atlantis-api
>   labels:
>     app: atlantis
>     chart: atlantis-4.21.1
>     helm.sh/chart: atlantis-4.21.1
>     release: atlantis
>     heritage: Helm
>   annotations:
>     aaa: bbb
> data:
>   apisecret: YWFh
24a43,44
>   annotations:
>     aaa: bbb
53a74,75
>   annotations:
>     aaa: bbb
75a98,100
>
>   annotations:
>     aaa: bbb
89,90c114,115
<         checksum/config: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
<         checksum/repo-config: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
---
>         checksum/config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
>         checksum/repo-config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
118a144,148
>           - name: ATLANTIS_API_SECRET
>             valueFrom:
>               secretKeyRef:
>                 name: atlantis-api
>                 key: apisecret
165a196,197
>   annotations:
>     aaa: bbb
```

## references

Closes https://github.com/runatlantis/helm-charts/issues/140

### CODE REVIEW

**Comment by GMartinez-Sisti** [comment]
@roytev I love these unit tests 😍 

```
FAIL  test statefulset	charts/atlantis/tests/statefulset_test.yaml
	- default values

		- asserts[12] `equal` fail
			Template:	atlantis/templates/statefulset.yaml
			DocumentIndex:	0
			Path:	spec.template.metadata.annotations
			Expected to equal:
				checksum/config: e3b0c44298fc1c149afbf4c8996fb92427ae41e[46](https://github.com/runatlantis/helm-charts/actions/runs/8072357303/job/22053924032?pr=356#step:8:47)49b934ca495991b7852b855
				checksum/repo-config: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
			Actual:
				checksum/config: 01ba[47](https://github.com/runatlantis/helm-charts/actions/runs/8072357303/job/22053924032?pr=356#step:8:48)19c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
				checksum/repo-config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c058ef8f9805daca546b
			Diff:
				--- Expected
				+++ Actual
				@@ -1,3 +1,3 @@
				-checksum/config: e3b0c44298fc1c1[49](https://github.com/runatlantis/helm-charts/actions/runs/8072357303/job/22053924032?pr=356#step:8:50)afbf4c8996fb92427ae41e4649b934ca495991b7852b855
				-checksum/repo-config: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
				+checksum/config: 01ba4719c80b6fe911b091a7c0[51](https://github.com/runatlantis/helm-charts/actions/runs/8072357303/job/22053924032?pr=356#step:8:52)24b64eeece964e09c058ef8f9805daca[54](https://github.com/runatlantis/helm-charts/actions/runs/8072357303/job/22053924032?pr=356#step:8:55)6b
				+checksum/repo-config: 01ba4719c80b6fe911b091a7c05124b64eeece964e09c0[58](https://github.com/runatlantis/helm-charts/actions/runs/8072357303/job/22053924032?pr=356#step:8:59)ef8f9805daca546b
```

File: charts/atlantis/templates/statefulset.yaml [hunk 1]
```diff
@@ -3,13 +3,18 @@ kind: StatefulSet
...
   name: {{ template "atlantis.fullname" . }}
   labels:
-{{- include "atlantis.labels" . | nindent 4 }}
-{{- if .Values.statefulSet.labels }}
-{{ toYaml .Values.statefulSet.labels | indent 4 }}
-{{- end }}
-  {{- with .Values.statefulSet.annotations }}
+    {{- include "atlantis.labels" . | nindent 4 }}
+    {{ with .Values.statefulSet.labels }}
+    {{- toYaml .Values.statefulSet.labels | nindent 4 }}
```
Comment (peikk0) [inline charts/atlantis/templates/statefulset.yaml:8]: This is broken, should be `toYaml .` instead.

Comment (GMartinez-Sisti) [reply charts/atlantis/templates/statefulset.yaml:8]: That's correct. Can you send a fix and we'll merge it pls?

Comment (GMartinez-Sisti) [reply charts/atlantis/templates/statefulset.yaml:8]: And `{{ with` should be `{{- with` to prevent the extra line too.

Comment (peikk0) [reply charts/atlantis/templates/statefulset.yaml:8]: Fix in #364

**Pull request merged**